src/object/srv_ec_aggregate.c

```diff
@@ -334,6 +404,153 @@ agg_sgl_fini(d_sg_list_t *sgl)
 	}
 }
 
+/* Determines if an extent overlaps a cell.
+ */
+static bool
+agg_overlap(unsigned int estart, unsigned int elen, unsigned int cell,
+	    unsigned int k, unsigned int len, daos_off_t stripenum)
+{
+	daos_off_t cell_start = k * len * stripenum + len * cell;
+
+	estart += k * len * stripenum;
+	if (cell_start <= estart && estart < cell_start + len)
+		return true;
+	if (estart <= cell_start && cell_start < estart + elen)
+		return true;
+	return false;
+}
+
+static unsigned int
+agg_count_cells(uint8_t *fcbit_map, uint8_t *tbit_map, unsigned int estart,
+		unsigned int elen, unsigned int k, unsigned int len,
+		unsigned int stripenum, unsigned int *full_cell_cnt)
+{
+	unsigned int i, cell_cnt = 0;
+
+	for (i = 0; i < k; i++) {
+		if (i * len >= estart &&  estart + elen >= (i + 1) * len) {
+			setbit(tbit_map, i);
+			if (full_cell_cnt)
+				D_ASSERT(fcbit_map);
+				setbit(fcbit_map, i);
+				(*full_cell_cnt)++;
+			cell_cnt++;
+		} else if (agg_overlap(estart, elen, i, k, len, stripenum)) {
+			setbit(tbit_map, i);
+			cell_cnt++;
+		}
+	}
+
+	return cell_cnt;
```

Comment:
These function seems general enough to be a general EC striping API, instead of only for agg.

Reply:
Could be.  If they're needed elsewhere they could easily be relocated in a later patch.